Cargo.toml

```diff
@@ -14,6 +14,17 @@ exclude = [
     "*.png",
 ]
 
+[dependencies]
+serde = { version = "1.0", optional = true, default-features = false, features = ["derive"] }
+
+# This is a dev-dependency, but those cannot be optional, so it's included here and not built by
+# default. It is used with the serde feature unit tests.
+serde_json = { version = "1", optional = true }
+
 [dev-dependencies]
+bmfont = { path = ".", features = ["serde", "serde_json"] }
```

Comment:
What? :)

Reply:
It's the first time I see the crate itself being a dependency.

Reply:
This allows `cargo test` to test the default features and also our feature set. This may be black magic. This technique could be extended with a secondary dependency to check the `parse-errors` feature too. Not sure how complicated this should get, at the benefit of zero user involvement while testing everything.

Reply:
Okay. Got it :)